src/pybind/mgr/dashboard/frontend/src/app/core/error/error.component.ts

```diff
@@ -0,0 +1,45 @@
+import { Location } from '@angular/common';
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+
+@Component({
+  selector: 'cd-error',
+  templateUrl: './error.component.html',
+  styleUrls: ['./error.component.scss']
+})
+export class ErrorComponent implements OnInit {
+  topHeader: string;
+  topMessage: string;
+  btmHeader: string;
+  btmMessage: string;
+  section: string;
+  section_info: string;
+  fa_icon: string;
+
+  constructor(public route: ActivatedRoute, private location: Location) {}
+
+  ngOnInit() {
+    try {
+      this.topMessage = history.state.topMessage;
+      this.topHeader = history.state.topHeader;
+      this.btmHeader = history.state.btmHeader;
+      this.btmMessage = history.state.btmMessage;
+      this.section = history.state.section;
+      this.section_info = history.state.section_info;
+      this.fa_icon = history.state.fa_icon;
```

Comment:
[`this.state = this.location.getState()`](https://angular.io/api/common/Location#getState)? I rather prefer the Angular scoped classes than the raw Javascript globals.

More details here: https://www.tektutorialshub.com/angular/angular-pass-data-to-route/#accessing-the-state-value

Reply:
I tried using the 'this.location.getState()' and the 'this.router.getCurrentNavigation().extras.state' but was receiving the result as undefined everytime..found some content related to this in this article https://medium.com/ableneo/how-to-pass-data-between-routed-components-in-angular-2306308d8255 ,so decided to continue with 'history.state'.